### PR TITLE
Created routes. Implemented protected Routes 

### PR DESCRIPTION
**Pull request by AlonaVladymyrovaTrinity**
This pull request implements protected routes to ensure that certain routes can only be accessed by authenticated users. The changes include:

- Updated the App component to create routs and manage user authentication status.

- Integrated the ProtectedRoute component which acts as a wrapper for protected routes, redirecting unauthenticated users to the login page.

- ProtectedRoute component was added for routes requiring authentication, such as "Profile," "UpdateProfile," and "UpdatePassword."

- Added new folders and files for routes:        

1. new file:   src/component/Adoption/ConfirmApplication.js       
2.  new file:   src/component/Adoption/FavoritesPets.js        
3. new file:   src/component/Donate/Donate.js        
4. new file:   src/component/NoFavorites/NoFavorites.js        
5. new file:   src/component/Pet/PetDetails.js        
6. new file:   src/component/Pet/Pets.js        
7. new file:   src/component/Pet/Search.js        
8. new file:   src/component/User/ForgotPassword.js        
9. new file:   src/component/User/LoginSignUp.js        
10. new file:   src/component/User/Profile.js        
11. new file:   src/component/User/ResetPassword.js        
12. new file:   src/component/User/UpdatePassword.js        
13. new file:   src/component/User/UpdateProfile.js        
14. new file:   src/component/layout/Contact/Contact.js        
15. new file:   src/component/layout/Footer/Footer.js

These changes enhance security, restrict unauthorized access, and improve the user experience.

### CODE REVIEW

**Review by LiubovCass — APPROVED** [review]
Good job. No questions

**Review by badgerbadgers — APPROVED** [review]
LGTM!

File: src/component/Adoption/ConfirmApplication.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const ConfirmApplicatio = () => {
```
Comment (OxiBo) [inline src/component/Adoption/ConfirmApplication.js]: There is a typo here in the component name - missing 'n' at the end

**Review by OxiBo — CHANGES_REQUESTED** [review]
Everything is looking good! Correct the typo and it is good to go to main to serve as a good skeleton. Personally I prefer a little different structure of files and folders: separating PAGES in /page folder from all other components. Current structure could be a bit confusing as far as quickly understanding what is a page component and what as a UI component for example. This  also works! Good job!

File: src/component/Adoption/ConfirmApplication.js (hunk 1 above)
Comment (AlonaVladymyrovaTrinity) [reply src/component/Adoption/ConfirmApplication.js]: OK! Thanks!

**Comment by AlonaVladymyrovaTrinity** [comment]
The typo has been successfully fixed.

**Review by OxiBo — APPROVED** [review]

**Pull request merged**